kyaml/kio/pkgio_reader.go

```diff
@@ -276,17 +295,102 @@ func (r *LocalPackageReader) initReaderAnnotations(path string, _ os.FileInfo) {
 // ShouldSkipDir returns a filepath.SkipDir if the directory should be skipped
 func (r *LocalPackageReader) ShouldSkipDir(path string) error {
 	if r.PackageFileName == "" {
+		// If the folder is not a package, but covered by the .krmignore file,
+		// we skip it.
+		if r.ignoreFilesMatcher.matchDir(path) {
+			return filepath.SkipDir
+		}
 		return nil
 	}
 	// check if this is a subpackage
 	_, err := os.Stat(filepath.Join(path, r.PackageFileName))
 	if os.IsNotExist(err) {
+		// Skip the folder if it is covered by the .krmignore file.
+		if r.ignoreFilesMatcher.matchDir(path) {
+			return filepath.SkipDir
+		}
 		return nil
 	} else if err != nil {
 		return errors.Wrap(err)
 	}
 	if !r.IncludeSubpackages {
 		return filepath.SkipDir
 	}
+	// We don't allow the .krmignore file in a package cause us to skip
+	// a subpackage. So if we have found a package file in the folder and
+	// we should include subpackages, we don't check the .krmignore file. We
+	// do however check whether the package contains a .krmignore file.
+	return r.ignoreFilesMatcher.readIgnoreFile(path)
+}
+
+// ignoreFilesMatcher keeps a stack of matchers for .krmignore files. So as
```

Comment:
Please make files ignorefilesmatcher.go and test, because it makes it easier to navigate / grok.

Also, the PR description is great, but almost nobody will see it. Can you work it into the text here please?

Reply:
Done